diff --git a/verl/models/mcore/util.py b/verl/models/mcore/util.py
index b5d16c48..364ddab7 100644
--- a/verl/models/mcore/util.py
+++ b/verl/models/mcore/util.py
@@ -117,12 +117,12 @@ def preprocess_packed_seqs(
 
     packed_seq_params = PackedSeqParams(
         qkv_format="thd",
-        cu_seqlens_q=cu_seqlens_padded,
+        cu_seqlens_q=cu_seqlens,
         max_seqlen_q=max_seqlen_in_batch,
-        cu_seqlens_kv=cu_seqlens_padded,
+        cu_seqlens_kv=cu_seqlens,
         max_seqlen_kv=max_seqlen_in_batch,
-        cu_seqlens_q_padded=cu_seqlens_padded,
-        cu_seqlens_kv_padded=cu_seqlens_padded,
+        cu_seqlens_q_padded=cu_seqlens,
+        cu_seqlens_kv_padded=cu_seqlens,
     )
     if pre_process:
         return input_ids_rmpad.unsqueeze(0), packed_seq_params
@@ -422,12 +422,12 @@ def preprocess_thd_no_padding(
 
     packed_seq_params = PackedSeqParams(
         qkv_format="thd",
-        cu_seqlens_q=cu_seqlens_padded,
+        cu_seqlens_q=cu_seqlens,
         max_seqlen_q=max_seqlen_in_batch,
-        cu_seqlens_kv=cu_seqlens_padded,
+        cu_seqlens_kv=cu_seqlens,
         max_seqlen_kv=max_seqlen_in_batch,
-        cu_seqlens_q_padded=cu_seqlens_padded,
-        cu_seqlens_kv_padded=cu_seqlens_padded,
+        cu_seqlens_q_padded=cu_seqlens,
+        cu_seqlens_kv_padded=cu_seqlens,
     )
     if pre_process:
         return input_ids_rmpad.unsqueeze(0), packed_seq_params, position_ids_rmpad.unsqueeze(0)
